### PR TITLE
Work around Julia 1.9.0 regression

### PR DESCRIPTION
**Pull request by YingboMa**
Ref: https://github.com/JuliaLang/julia/issues/49766

### CODE REVIEW

File: src/structural_transformation/bareiss.jl [hunk 1]
```diff
@@ -143,6 +143,9 @@ end
     V = M[(k + 1):end, (k + 1):end]
     V .= exactdiv.(V .* pivot .- M[(k + 1):end, k] * M[k, (k + 1):end]', prev_pivot)
     zero!(M, (k + 1):size(M, 1), k)
+    if M isa AbstractSparseMatrixCSC
+        dropzeros!(M)
```
Comment (Keno) [inline src/structural_transformation/bareiss.jl:147]: Why is this required are we relying on structural rather than value comparisons somewhere?

Comment (YingboMa) [reply src/structural_transformation/bareiss.jl:147]: There's another fun bug https://github.com/JuliaLang/julia/issues/49771
